bool auto_shoot
bool armor_detected